### ISSUE TITLE
remove native-too support from Cabal?

### ISSUE THREAD

**Issue by luite** [ghcjs/ghcjs#724]
`native-too` mode is a way of having GHCJS build both native code and JavaScript code for the same Haskell packages. It used to be a requirement to have native code for Template Haskell and custom Cabal setup scripts. Later native code became optional, but some people were still using this option for something else, hence the `native-too` flag.

I'd like to work towards a real "web" platform for our compiler to target. GHCJS should be just a cross compiler for that target, and I don't think we really need the `native-too` trick anymore. It hasn't really been supported in GHCJS for a long time anyway.

Native-too depends on all our `configure` scripts and other build tools running on the host platform context and returning correct answers for that. I've patched some source files to be compatible with `native-too` by having preprocessor conditionals in the output (of `hsc2hs` files for example). But this has always felt like a stopgap solution.

Cabal support for GHCJS has fallen behind GHC a little bit, and I'm working on adding the latest updates to the GHCJS compiler flavour. Does anyone have objections to completely dropping `native-too` support in Cabal 2.6?

**Comment by Ericson2314** [comment]
Yes please drop. I've taught nixpkgs how to use the native GHC to build Setup.hs, and cabal-install should also learn that. [I've been drip drop working on this for years, but no time to finish myself.]

**Comment by hvr** [comment]
@Ericson2314 you mean, cabal should learn how to better deal with cross-compilation... and properly separate between compiling stuff for the build-host (which can executed by cabal) vs stuff for the target-host (which generally is not always possible to execute)... :-)

**Comment by Ericson2314** [comment]
Yes indeed :), along with making Setup.hs a proper (executable with funny name?) Component.
